### PR TITLE
Change maybe_promote fill_value to dt64/td64 NaT instead of iNaT

### PR DESCRIPTION
**Pull request by jbrockmendel**
Unambiguous, preliminary for fixing more xfails.

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -350,9 +350,14 @@ def maybe_promote(dtype, fill_value=np.nan):
 
     # returns tuple of (dtype, fill_value)
     if issubclass(dtype.type, np.datetime64):
-        fill_value = tslibs.Timestamp(fill_value).value
+        fill_value = tslibs.Timestamp(fill_value).to_datetime64()
     elif issubclass(dtype.type, np.timedelta64):
-        fill_value = tslibs.Timedelta(fill_value).value
+        fv = tslibs.Timedelta(fill_value)
+        if fv is NaT:
+            # NaT has no `to_timedelta6` method
```
Comment (jreback) [inline pandas/core/dtypes/cast.py]: really? should we add one instead of this?

File: pandas/core/dtypes/cast.py [hunk 2]
```diff
@@ -393,7 +398,7 @@ def maybe_promote(dtype, fill_value=np.nan):
             dtype = np.float64
             fill_value = np.nan
         elif is_datetime_or_timedelta_dtype(dtype):
```
Comment (jreback) [inline pandas/core/dtypes/cast.py]: would it be better to branch this to be explict (e.g. datetime and timedelta branches)

Comment (jreback) [reply pandas/core/dtypes/cast.py]: actually why do we have a branch for a subclass of np.datetime64 AND this one (and same for timedelta)?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/dtypes/cast.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py]: NaT is pretty specifically a datetime.  I think adding this there would muddy the waters

Comment (jreback) [reply pandas/core/dtypes/cast.py]: ok

File: pandas/core/dtypes/cast.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py]: not clear.  my current plan is to do a couple more small PRs of xfail-fixing before getting into refactoring

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok thanks @jbrockmendel 

yeah this for sure needs refactoring, but ok to un-xfail tests first

**Pull request merged**